client/src/partials/patients/registry/registry.js

```diff
@@ -14,60 +14,68 @@ PatientRegistryController.$inject = [
  */
 function PatientRegistryController($translate, Patients, $uibModal) {
   var vm = this;
-  // options for the UI grid
   
   vm.search = search;
   vm.momentAge = momentAge;
-    
+ 
+  // track if module is making a HTTP request for patients
+  vm.loading = false;
+  
   /** TODO MANAGE COLUMN : LAST_TRANSACTION */
+  // the column attribute `displayName` must be used in favour of `name` in order to allow `headerCellFilter` to function
   vm.uiGridOptions = {
-    appScopeProvider : vm, // ensure that the controller's `this` variable is bound to appScope
+    appScopeProvider : vm,
     enableColumnMenus : false,
+    headerCellFilter: 'translate',
     columnDefs : [
-      { field : 'reference', name : $translate.instant('TABLE.COLUMNS.REFERENCE') },
-      { field : 'patientName', name : $translate.instant('TABLE.COLUMNS.NAME') },
-      { field : 'patientAge', name : $translate.instant('TABLE.COLUMNS.AGE') },
-      { field : 'sex', name : $translate.instant('TABLE.COLUMNS.GENDER') },
-      { field : 'hospital_no', name : $translate.instant('TABLE.COLUMNS.HOSPITAL_FILE_NR') },
-      { field : 'registration_date', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.DATE_REGISTERED') },
-      { field : 'last_visit', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.LAST_VISIT') },
-      { field : 'dob', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.DOB') }  
+      { field : 'reference', displayName : 'TABLE.COLUMNS.REFERENCE', headerCellFilter: 'translate' },
+      { field : 'patientName', displayName : 'TABLE.COLUMNS.NAME', headerCellFilter : 'translate' },
+      { field : 'patientAge', displayName : 'TABLE.COLUMNS.AGE', headerCellFilter : 'translate'  },
+      { field : 'sex', displayName : 'TABLE.COLUMNS.GENDER', headerCellFilter : 'translate'  },
+      { field : 'hospital_no', displayName : 'TABLE.COLUMNS.HOSPITAL_FILE_NR', headerCellFilter : 'translate'  },
+      { field : 'registration_date', cellFilter:'date', displayName : 'TABLE.COLUMNS.DATE_REGISTERED', headerCellFilter : 'translate'  },
+      { field : 'last_visit', cellFilter:'date', displayName : 'TABLE.COLUMNS.LAST_VISIT', headerCellFilter : 'translate'  },
+      { field : 'dob', cellFilter:'date', displayName : 'TABLE.COLUMNS.DOB', headerCellFilter : 'translate'  },
+      { name : 'Actions', displayName : '', cellTemplate : '<div style="padding : 5px"><a ui-sref="patientRecord({patientID : row.entity.uuid})"><span class="glyphicon glyphicon-list-alt"></span> Record</a> <a ui-sref="patientEdit({uuid : row.entity.uuid})"><span class="glyphicon glyphicon-edit"></span> Edit</a></div>'}
     ],
     enableSorting : true
   };
-
+  
   // load Patient Registry Grid
   function loadGrid() {
+    vm.loading = true;
     Patients.read().then(function (patients) {
+      vm.loading = false;
```

Comment:
This should be in the `finally()` block of the promise - if an error occurs (which it shouldn't), it will load indefinitely.